android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUILabel.java

```diff
@@ -88,99 +99,176 @@ protected void onLayout(boolean changed, int left, int top, int right, int botto
-			    }
+				}
+				return super.onTouchEvent(event);
+			}
 		};
 		tv.setGravity(Gravity.CENTER_VERTICAL | Gravity.LEFT);
 		tv.setPadding(0, 0, 0, 0);
-		tv.setInputType(InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_FLAG_MULTI_LINE);
-		tv.setKeyListener(null);
 		tv.setFocusable(false);
-		tv.setSingleLine(false);
-		tv.setEllipsize(ellipsize);
+		tv.setEllipsize(this.ellipsize);
+		tv.setSingleLine(!this.wordWrap);
 		TiUIHelper.styleText(tv, null);
-		defaultColor =  tv.getCurrentTextColor();
+		this.unscaledFontSizeInPixels = tv.getTextSize();
+		this.defaultColor =  tv.getCurrentTextColor();
 		setNativeView(tv);
-
 	}
-	
+
 	/**
 	 * Method used to decrease the fontsize of the text to fit the width
 	 * fontsize should be >= than the property minimumFontSize
-	 * @param v
+	 * @param view
 	 */
-	private void adjustTextFontSize(View v){	
-		if (minimumFontSize != null){
-			TextView tv = (TextView) v;
-
-			if (tv != null) {
-				if (autoshrinkSetFontSize != null) {
-					if (tv.getTextSize() == TiConvert.toFloat(autoshrinkSetFontSize)) {
-						String[] fontProperties = TiUIHelper.getFontProperties(proxy.getProperties());
-
-						if (fontProperties.length > TiUIHelper.FONT_SIZE_POSITION && fontProperties[TiUIHelper.FONT_SIZE_POSITION] != null) {
-							tv.setTextSize(TiUIHelper.getSizeUnits(fontProperties[TiUIHelper.FONT_SIZE_POSITION]), TiUIHelper.getSize(fontProperties[TiUIHelper.FONT_SIZE_POSITION]));
-						} else {
-							tv.setTextSize(TiUIHelper.getSizeUnits(null), TiUIHelper.getSize(null));
-						}
-					}
-				}
-			    
-			    TextPaint textPaint = tv.getPaint();
-				if (textPaint != null) {
-					float stringWidth = textPaint.measureText(tv.getText().toString());
-					int textViewWidth = tv.getWidth();
-					if (textViewWidth < stringWidth && stringWidth > 0) {
-						float fontSize = (textViewWidth / stringWidth) * tv.getTextSize();
-						autoshrinkSetFontSize = fontSize > TiConvert.toFloat(minimumFontSize, 0) ? String.valueOf(fontSize) : minimumFontSize;
-						tv.setTextSize(TiUIHelper.getSizeUnits(autoshrinkSetFontSize), TiUIHelper.getSize(autoshrinkSetFontSize));
-					}
-				}
+	private void adjustTextFontSize(View view)
+	{
+		// Do not continue if auto-sizing the font is disabled.
+		if (this.minimumFontSizeInPixels < FONT_SIZE_EPSILON) {
+			return;
+		}
+
+		// Make sure the given view is a TextView.
+		if ((view instanceof TextView) == false) {
+			return;
+		}
+		TextView textView = (TextView)view;
+
+		// Fetch the display's density scale and calculate pixel's per virtual point, rounded up.
+		// Note: To keep it simple, don't let the scale be below 1x.
+		//       Only obsolete "ldpi" devices use scales lower than this.
+		float densityScale = view.getResources().getDisplayMetrics().density;
+		if (densityScale <= 0) {
+			densityScale = 1.0f;
+		}
+
+		// Calculate the pixel width within the view that the text will be rendered to.
+		float viewContentWidth;
+		{
```

Comment:
I know, but the purpose of anonymous inner blocks is to restrict the variable scope and should be used for specific use cases instead of formatting.